diff --git a/drivers/net/wireless/aic8800/usb/aic8800_fdrv/rwnx_tx.c b/drivers/net/wireless/aic8800/usb/aic8800_fdrv/rwnx_tx.c
index ff8daa6b..7b3cb19e 100644
--- a/drivers/net/wireless/aic8800/usb/aic8800_fdrv/rwnx_tx.c
+++ b/drivers/net/wireless/aic8800/usb/aic8800_fdrv/rwnx_tx.c
@@ -321,8 +321,7 @@ u16 rwnx_select_txq(struct rwnx_vif *rwnx_vif, struct sk_buff *skb)
     }
     case NL80211_IFTYPE_MONITOR:
         skb->priority = TID_MGT;
-        txq = rwnx_txq_vif_get(rwnx_vif, NX_UNK_TXQ_TYPE);
-        netdev_queue = txq->ndev_idx;
+        netdev_queue = nx_bcmc_txq_ndev_idx;
         return netdev_queue;
     case NL80211_IFTYPE_AP_VLAN:
         if (rwnx_vif->ap_vlan.sta_4a) {
@@ -2233,7 +2232,7 @@ int rwnx_txdatacfm(void *pthis, void *host_id)
     rwnx_txq_confirm_any(rwnx_hw, txq, hwq, sw_txhdr);
 
     /* Update txq and HW queue credits */
-    if (sw_txhdr->desc.host.flags & TXU_CNTRL_MGMT) {
+    if ((sw_txhdr->desc.host.flags & TXU_CNTRL_MGMT) && !sw_txhdr->raw_frame) {
         trace_printk("done=%d retry_required=%d sw_retry_required=%d acknowledged=%d\n",
                      rwnx_txst.tx_done, rwnx_txst.retry_required,
                      rwnx_txst.sw_retry_required, rwnx_txst.acknowledged);
